src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompilerCache.cs

```diff
@@ -67,27 +79,26 @@ public CompilerCache(IFileProvider fileProvider)
         /// <inheritdoc />
         public CompilerCacheResult GetOrAdd(
             string relativePath,
-            Func<string, CompilerCacheContext> cacheContextFactory)
+            Func<string, CompilationResult> compile)
```

Comment:
Weird design - where the compiler cache did not know about the compiler. It might be cleaner to change this so that cache just calls in to the compiler from here.